PaddleCV/PaddleDetection/tools/train.py

```diff
@@ -136,15 +148,42 @@ def main():
     build_strategy.enable_inplace = False
     sync_bn = getattr(model.backbone, 'norm_type', None) == 'sync_bn'
     # only enable sync_bn in multi GPU devices
-    build_strategy.sync_batch_norm = sync_bn and devices_num > 1 and cfg.use_gpu
-    train_compile_program = fluid.compiler.CompiledProgram(
-        train_prog).with_data_parallel(
-            loss_name=loss.name, build_strategy=build_strategy)
+    build_strategy.sync_batch_norm = sync_bn and devices_num > 1 \
+        and cfg.use_gpu
+
+    if FLAGS.dist:
+        trainer_id = int(env['PADDLE_TRAINER_ID'])
+        trainers = env['PADDLE_TRAINER_ENDPOINTS']
+        current_endpoint = env['PADDLE_CURRENT_ENDPOINT']
+        num_trainers = int(env['PADDLE_TRAINERS_NUM'])
+
+        config = fluid.DistributeTranspilerConfig()
+        config.mode = "nccl2"
+        t = fluid.DistributeTranspiler(config=config)
+        t.transpile(trainer_id, trainers=trainers,
+                    startup_program=startup_prog,
+                    current_endpoint=current_endpoint)
+
+        exe.run(startup_prog)
+
+        pe = fluid.ParallelExecutor(
+            use_cuda=True,
+            main_program=train_prog,
+            loss_name=loss.name,
+            num_trainers=num_trainers,
+            trainer_id=trainer_id)
+
+    else:
+        exe.run(startup_prog)
+        pe = fluid.ParallelExecutor(
+            main_program=train_prog,
+            use_cuda=True,
+            loss_name=loss.name,
+            build_strategy=build_strategy)
+
```

Comment:
需要参考image_classification里的代码简化这块的逻辑

Reply:
The model guide version does not  work, I suggest we merge this for now since it is working code. The style issue could be addressed later after all the major changes have been merged.